content/Snakemake_intro.md

```diff
@@ -12,27 +26,471 @@ Example usecase from GUI -> multiple scripts run manually -> bash script to run
+
+## Workflow reproducibility
+
+Reproducibility in computational workflows ensures that an analysis can be consistently repeated with the same results, which is crucial for validating findings, ensuring transparency, and facilitating collaboration. It allows other researchers to verify results, understand methodologies, and build upon previous work.
+
+### Project structure 
+
+Snakemake provides recommendations regarding project structure. 
+```
+├── .gitignore
+├── README.md
+├── LICENSE.md
+├── workflow
+│   ├── rules
+|   │   ├── module1.smk
+|   │   └── module2.smk
+│   ├── envs
+|   │   ├── tool1.yaml
+|   │   └── tool2.yaml
+│   ├── scripts
+|   │   ├── script1.py
+|   │   └── script2.R
+│   ├── notebooks
+|   │   ├── notebook1.py.ipynb
+|   │   └── notebook2.r.ipynb
+│   ├── report
+|   │   ├── plot1.rst
+|   │   └── plot2.rst
+|   └── Snakefile
+├── config
+│   ├── config.yaml
+│   └── some-sheet.tsv
+├── results
+└── resources
+```
+
+Following the [documentation](https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html#distribution-and-reproducibility): 
+> The workflow code goes into a subfolder workflow, while the configuration is stored in a subfolder config. Inside of the workflow subfolder, the central Snakefile marks the entrypoint of the workflow (it will be automatically discovered when running snakemake from the root of above structure).
+
+> Workflows set up in above structure can be easily used and combined via the Snakemake module system. Such deployment can even be automated via Snakedeploy. Moreover, by publishing a workflow on Github and following a set of additional rules the workflow will be automatically included in the Snakemake workflow catalog, thereby easing discovery and even automating its usage documentation.
+
+Following this structures enables some of the features, as well as enhances reproducibility.
+
+### Containerization
+
+Containers, like Docker, provide a lightweight and portable way to package and run applications. They encapsulate an application and its dependencies, ensuring it runs consistently across different environments. Unlike virtual machines, containers share the host system's kernel but isolate the application's processes, filesystem, and resources. This makes containers more efficient in terms of performance and resource usage, allowing for quick startup times and easy scalability. Docker is a popular containerization platform that simplifies creating, deploying, and managing containers, making it an essential tool for modern software development and deployment.
+
+To use containers with Snakemake, we add the `container` keyword inside a rule:
+
+```bash
+rule copy:
+    container:
+        "docker://alpine:3.14"
+    input:
+	    "hello.txt"
+    output: 
+	    "hello_copy.txt"
+    shell:
+	    "cp {input} {output}"
+```
+
+To execute that Snakefile, we have to add the `--software-deployment-method` flag:
+```bash
+snakemake --cores 1 --software-deployment-method apptainer
```

Comment:
Could it be docker here as well? If so then better to change it, to make it easier to follow (as the image in the Snakefile comes from docker(hub)).

Reply:
I will remove any docker mention before that